File: Benchmarking/stats/euler12/euler12.java
public class euler12
{
	public static void main(String[] args)
	{
		int numDiv = Integer.parseInt(args[0]);
		System.out.println(triangular(numDiv));
	}
	
	private static int divisors(int n)
	{
		int numFactors = 0;
		int ceiling = 2147483647;
		for(int i = 1; i <= (int)Math.ceil(Math.sqrt(n)+1); i++)
		{
			if(n % i == 0)
			{
				numFactors += 2;
			}
			if (i * i == n)
			{
				numFactors -= 1;
			}
		}
		return numFactors;
	}
	
	private static int triangular(int numDiv)
	{
		int ceil = 32767;
		int count;
		int k;
		for(int n = 1; n <= ceil; n++)
		{
			k = (int)n * (n + 1) / 2;
			if(n % 2 == 0)
			{
				count = divisors(n / 2) * divisors(n + 1);
			}
			else
			{
				count = divisors(n) * divisors((n + 1) / 2);
			}
			if(count > numDiv)
			{
				return k;
			}
		}
		return 0;
	}
}

File: Benchmarking/stats/euler12/euler12.cpp
//
//  main.cpp
//  Euler12
//
//  Created by Sonni on 20/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>
#include <limits>
#include <math.h>
#include <vector>
#include <set>
#include <map>

std::vector<int> prime_factors(unsigned int n)
{
    std::vector<int> factors;
    unsigned int d = 2;
    while (n > 1)
    {
        while (n % d == 0)
        {
            factors.push_back(d);
            n /= d;
        }
        d++;
    }
    return factors;
}

unsigned int triangular(unsigned int div)
{
    unsigned int limit = std::numeric_limits<int16_t>::max();
    unsigned int i = 0;
    for (unsigned int n = 1; n <= limit; n++)
    {
        i += n;
        unsigned int countDiv = 1;
        std::vector<int> fact = prime_factors(i);
        std::vector<int> distinct = fact;

        std::sort(distinct.begin(), distinct.end()); // can't do this, 
        std::vector<int>::iterator it = std::unique(distinct.begin(), distinct.end());
        distinct.erase(it, distinct.end());
       
        
        std::vector<unsigned int> countExp;
        std::map<int, int> dict;

        for (unsigned int i = 0; i < distinct.size(); i++)
        {
            for (unsigned int j = 0; j < fact.size(); j++)
            {
                if (distinct[i] == fact[j])
                {
                    if (dict.count(distinct[i]) != 0)
                    {
                        dict[distinct[i]] = dict[distinct[i]] + 1;
                    }
                    else
                        dict[distinct[i]] = 1;
                }
            }
        }
        
        
        for(std::map<int,int>::iterator it = dict.begin(); it != dict.end(); ++it)
        {
            countDiv *= dict[it->first] + 1;
        }
        if (countDiv > div)
            return i;
    }
    return -1;
}

int main(int argc, const char * argv[]) {
    triangular(atoi(argv[1]));
    return 0;
}
